eng/Versions.props

```diff
@@ -22,7 +22,7 @@
     <!-- Use the correct compiler version -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
     <!-- Dependencies from https://github.com/dotnet/roslyn -->
-    <MicrosoftNETCoreCompilersPackageVersion>3.8.0</MicrosoftNETCoreCompilersPackageVersion>
+    <MicrosoftNETCoreCompilersPackageVersion>3.9.0-2.final</MicrosoftNETCoreCompilersPackageVersion>
```

Comment:
@Evangelink I would suggest you make this change even in https://github.com/dotnet/roslyn-analyzers/pull/4555 to ensure none of the IDE warning violations are missed from being fixed.

Reply:
⚠️ This will make the minimum version of VS for working on this repository 16.9 Preview 2

Reply:
Hmm, I am fine to mark this PR as blocked till 16.9 public release.

I understand this requirement on Roslyn repo due to large contributor base. I think that is an acceptable requirement on this repo though, but don't see any real rush in doing this.

Reply:
Thanks for the warning @sharwell